kong/utils.go

```diff
@@ -797,32 +799,162 @@ func fillConfigRecordDefaultsAutoFields(plugin *Plugin, schema map[string]interf
+		}
+
+		p.Path = String(defaultPartialPathArray[0].String())
+	}
+
+	return nil
+}
+
+func fillPartialConfigsInPlugin(plugin *Plugin, partials []*Partial) (Configuration, error) {
+	config := plugin.Config.DeepCopy()
+
+	// internal function to find a partial by its ID
+	// this is useful to get partial's type for finding
+	// its default path in the config
+	findPartialByID := func(partialID string) *Partial {
+		for _, p := range partials {
+			if *p.ID == partialID {
+				return p
+			}
+		}
+		return nil
+	}
+
+	const configString string = "config."
+
+	for _, partialLink := range plugin.Partials {
+		sanitisedPath := strings.ReplaceAll(*partialLink.Path, configString, "")
+		partialNeeded := findPartialByID(*partialLink.ID)
+		if partialNeeded == nil {
+			return nil, fmt.Errorf("partial with ID %s not found", *partialLink.ID)
+		}
+		config[sanitisedPath] = partialNeeded.Config
+	}
+
+	return config, nil
+}
+
+func fillConfigRecordWithPartialsConfig(plugin *Plugin, pluginSchema map[string]interface{},
+	partials []*Partial,
+) error {
+	jsonb, err := json.Marshal(&pluginSchema)
+	if err != nil {
+		return err
+	}
+	gjsonSchema := gjson.ParseBytes((jsonb))
+
+	if plugin.Config == nil {
+		plugin.Config = make(Configuration)
+	}
+
+	if len(partials) > 0 {
+		err = getDefaultPartialPath(plugin.Partials, gjsonSchema, partials)
+		if err != nil {
+			return err
+		}
+		plugin.Config, err = fillPartialConfigsInPlugin(plugin, partials)
+		if err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
 // FillPluginsDefaults ingests plugin's defaults from its schema.
 // Takes in a plugin struct and mutate it in place.
 func FillPluginsDefaults(plugin *Plugin, schema Schema) error {
-	return fillConfigRecordDefaultsAutoFields(plugin, schema, FillRecordOptions{
+	return fillConfigRecordDefaultsAutoFields(plugin, schema, nil, FillRecordOptions{
+		FillDefaults: true,
+		FillAuto:     true,
+	})
+}
+
+// FillPluginsDefaultsWithPartials ingests plugin's defaults from its schema.
+// Takes in a plugin struct and mutate it in place.
+// It also fills in partial config
+func FillPluginsDefaultsWithPartials(plugin *Plugin, schema Schema, partials []*Partial) error {
+	return fillConfigRecordDefaultsAutoFields(plugin, schema, partials, FillRecordOptions{
 		FillDefaults: true,
 		FillAuto:     true,
 	})
 }
 
 // same as FillPluginsDefaults but allows configuring whether to fill defaults and auto fields.
 func FillPluginsDefaultsWithOpts(plugin *Plugin, schema map[string]interface{}, opts FillRecordOptions) error {
-	return fillConfigRecordDefaultsAutoFields(plugin, schema, opts)
+	return fillConfigRecordDefaultsAutoFields(plugin, schema, nil, opts)
+}
+
+// FillPluginsDefaultsWithPartials ingests plugin's defaults from its schema.
+// Takes in a plugin struct and mutate it in place.
+// It also fills in partial config
+func FillPluginWithPartials(plugin *Plugin, pluginSchema map[string]interface{}, partials []*Partial) error {
```

Comment:
Could we add simple UTs for `func`s added in this PR?

At the very least I think we should test the export functions like `FillPluginWithPartials` and `FillPluginsDefaultsWithPartials`